gh-121834: Improve `complex` C-API docs (#121835)
python/cpython: Doc/c-api/complex.rst

   The C structure which corresponds to the value portion of a Python complex
   number object.  Most of the functions for dealing with complex number objects
   use structures of this type as input or output values, as appropriate.

>    .. c:member:: double real
>    .. c:member:: double imag
> 
>       It is defined as::
> 
>          typedef struct {
>             double real;
>             double imag;
>          } Py_complex;

Reviewer: Maybe just drop typedef?

```suggestion
   .. c:member:: double real

      real part

   .. c:member:: double imag
   
      imaginary part

```

Reviewer 2: Yes, though with a complete sentence as the description of the two.

Author: I like that it shows all the details. Since it is rather compact, I would prefer to keep it.

>    .. c:member:: double real
>    .. c:member:: double imag
> 
>       It is defined as::
> 
>          typedef struct {
>             double real;
>             double imag;
>          } Py_complex;

Reviewer 3: That will render OK, but semantically it attaches the code snippet to `imag`.
To get entries for `real` and `imag` without a description, followed by more detail for `Py_complex`, the syntax is:
```suggestion
   .. c:member:: double real
                 double imag

   The structure is defined as::

      typedef struct {
          double real;
          double imag;
      } Py_complex;
```

   The C structure which corresponds to the value portion of a Python complex
   number object.  Most of the functions for dealing with complex number objects
   use structures of this type as input or output values, as appropriate.

>    .. c:namespace:: Py_complex

Reviewer: ```suggestion
   .. c:type:: Py_complex
```

   :meth:`~object.__complex__` method, this method will first be called to
   convert *op* to a Python complex number object.  If :meth:`!__complex__` is
   not defined then it falls back to call :c:func:`PyFloat_AsDouble` and
   returns ``0.0`` on success.

   Upon failure, this method returns ``-1.0`` with an exception set, so one

>    should call :c:func:`PyErr_Occurred` to check for errors.

Reviewer: Oops, this shouldn't be backported to 3.12.   This requires #109647.

Author: This can be a partial backport, no problem.

   method, this method will first be called to convert *op* to a Python complex
   number object.  If :meth:`!__complex__` is not defined then it falls back to
   :meth:`~object.__float__`.  If :meth:`!__float__` is not defined then it falls back
   to :meth:`~object.__index__`.

   Upon failure, this method returns :c:type:`Py_complex`

>    with ``.real`` set to ``-1.0`` and with an exception set, so one

Reviewer: Nit:
```suggestion
   with :c:member:`~Py_complex.real` set to ``-1.0`` and with an exception set, so one
```
